app/src/main/java/pl/droidevs/books/library/BookActivity.java

```diff
@@ -0,0 +1,158 @@
+    ImageView imageView;
+
+    @BindView(R.id.collapsing_toolbar_layout)
+    CollapsingToolbarLayout collapsingToolbarLayout;
+
+    @BindView(R.id.shadow_view)
+    View shadowView;
+
+    @BindView(R.id.author_tv)
+    TextView authorTextView;
+
+    @BindView(R.id.category_tv)
+    TextView categoryTextView;
+
+    @BindView(R.id.description_tv)
+    TextView descriptionTextView;
+    //endregion
+
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        getSupportActionBar().hide();
+        setContentView(R.layout.activity_book);
+
+        AndroidInjection.inject(this);
+        ButterKnife.bind(this);
+
+        if (savedInstanceState != null) {
+            this.animationBundle = savedInstanceState.getBundle(BUNDLE_EXTRAS);
+        } else {
+            this.animationBundle = getIntent().getBundleExtra(BUNDLE_EXTRAS);
+        }
+
+        setupAnimations();
+        setupViewModel();
+    }
+
+    void setupAnimations() {
+        imageView.setTransitionName(animationBundle.getString(EXTRAS_IMAGE_TRANSITION_NAME));
+        //TODO title animation: animationBundle.getString(EXTRAS_TITLE_TRANSITION_NAME);
+        authorTextView.setTransitionName(animationBundle.getString(EXTRAS_AUTHOR_TRANSITION_NAME));
+        shadowView.setTransitionName(animationBundle.getString(EXTRAS_SHADOW_TRANSITION_NAME));
+    }
+
+    private void setupViewModel() {
+        viewModel = ViewModelProviders
+                .of(this, viewModelFactory)
+                .get(BookViewModel.class);
+        viewModel.setBookId((BookId) getIntent().getSerializableExtra(EXTRAS_BOOK_ID));
+        viewModel.getBook().observe(this, book -> {
+
+            if (book != null) {
+                setupBookViews(book);
+            }
+        });
+    }
+
+    void setupBookViews(Book book) {
+        collapsingToolbarLayout.setTitle(book.getTitle());
+        authorTextView.setText(book.getAuthor());
+        categoryTextView.setText(book.getCategory().toString());
+        descriptionTextView.setText(book.getDescription());
+
+        if (book.getImageUrl() != null) {
+            loadCover(book.getImageUrl());
+        }
+    }
+
+    void loadCover(String imageUrl) {
+        Glide.with(this)
+                .asBitmap()
+                .load(imageUrl)
+                .into(new BitmapImageViewTarget(imageView) {
+                    @Override
+                    public void onResourceReady(Bitmap resource, @Nullable Transition<? super Bitmap> transition) {
+                        super.onResourceReady(resource, transition);
+
+                        setupBarColors(resource);
+                    }
+                });
+    }
+
+    void setupBarColors(Bitmap resource) {
+        int[] colors = getColorsForBarsFromBitmap(resource);
+
+        if (colors != null) {
+            collapsingToolbarLayout.setContentScrimColor(colors[0]);
```

Comment:
Jeśli odwołujesz się do elementu tablicy o indeksie 0 i 1 to tak naprawdę nie zauważyłeś obiektu, który powinien posiadać dwa pola o znaczących nazwach.